### PR TITLE
Host release prep 4.1036.0

### PR DESCRIPTION
**Pull request by v-imohammad**
Updated release notes and Host version to 4.1036.0

### Pull request checklist

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [x] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

**Review by fabiocav — APPROVED** [review]

**Review by Francisco-Gamino — COMMENTED** [review]
LGTM

**Review by Francisco-Gamino — APPROVED** [review]
LGTM

File: release_notes.md [hunk 1]
```diff
@@ -16,5 +16,8 @@
 - Update Java Worker Version to [2.15.0](https://github.com/Azure/azure-functions-java-worker/releases/tag/2.15.0)
   - Update grpc-protobuf to 1.64.0 and application insights agent version to 3.5.2
 - Resolved thread safety issue in the `GrpcWorkerChannel.LoadResponse` method. (#10352)
+- Fixing thread safety issue in the GrpcWorkerChannel.LoadResponse method (#10363)
```
Comment (Francisco-Gamino) [inline release_notes.md:19]: Remove this.

File: release_notes.md [hunk 2]
```diff
@@ -16,5 +16,8 @@
...
 - Worker termination path updated with sanitized logging (#10367)
+- Avoid redundant DiagnosticEvents error message (#10395)
 - Added logic to shim older versions of the .NET Worker JsonFunctionProvider to ensure backwards compatibility (#10410)
+- Shimming legacy .NET worker JsonFunctionProvider to ensure backwards compatibility (#10411)
```
Comment (Francisco-Gamino) [inline release_notes.md:23]: Remove this line.

**Pull request merged**